FhirToDataLake/src/Microsoft.Health.AnalyticsConnector.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -212,7 +247,136 @@ await foreach (FhirToDataLakeProcessingJobInputData input in inputs.WithCancella
+                    throw new SynapsePipelineInternalException($"Initialize job status failed in orchestrator job {_jobInfo.Id}.");
+                }
+
+                jobStatus = await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, _jobInfo.Id, cancellationToken);
+            }
+
+            return jobStatus;
+        }
+
+        private async Task<OrchestratorJobStatusEntity> UpdateJobStatusWithResultAsync(CancellationToken cancellationToken)
+        {
+            _jobStatus.StatisticResult = JsonConvert.SerializeObject(_result);
+
+            if (!await _metadataStore.TryUpdateEntityAsync(_jobStatus))
+            {
+                _logger.LogInformation("Update orchestrator job status failed in job {0}.", _jobInfo.Id);
+                throw new SynapsePipelineInternalException($"Update orchestrator job status failed in job  {_jobInfo.Id}.");
+            }
+
+            return await _metadataStore.GetOrchestratorJobStatusAsync(_jobInfo.QueueType, _jobInfo.GroupId, _jobInfo.Id, cancellationToken);
+        }
+
         private async IAsyncEnumerable<FhirToDataLakeProcessingJobInputData> GetInputsAsyncForSystem([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // Resume the submitting job.
+            if (_result.SubmittingProcessingJob != null)
+            {
+                foreach (var job in _result.SubmittingProcessingJob)
+                {
+                    yield return JsonConvert.DeserializeObject<FhirToDataLakeProcessingJobInputData>(job);
+                }
+            }
+
+            List<TypeFilter> typeFilters = await _filterManager.GetTypeFiltersAsync(cancellationToken);
+            var resourceTypes = typeFilters.Select(filter => filter.ResourceType).ToList();
+
+            // Split jobs by resource types.
+            foreach (var resourceType in resourceTypes)
+            {
+                var startTime = _result.SubmittedResourceTimestamps.ContainsKey(resourceType) ? _result.SubmittedResourceTimestamps[resourceType] : _inputData.DataStartTime;
+
+                // The resource type has already been processed.
+                if (startTime >= _inputData.DataEndTime)
+                {
+                    continue;
+                }
+
+                IAsyncEnumerable<SubJobInfo> subJobs = _jobSpliter.SplitJobAsync(resourceType, startTime, _inputData.DataEndTime, cancellationToken);
+
+                await foreach (SubJobInfo subJob in subJobs.WithCancellation(cancellationToken))
+                {
+                    if (subJob.ResourceCount == 0)
+                    {
+                        _result.SubmittedResourceTimestamps[resourceType] = subJob.TimeRange.DataEndTime;
+                        continue;
+                    }
+                    else if (subJob.ResourceCount < LowBoundOfProcessingJobResourceCount)
+                    {
+                        // Small size job, put it into candidate list.
```

Comment:
If orchestrator job fails, can we recover the small job pool?

Reply:
The small job will be pushed into pool again.